### PR TITLE
Add Fortran test for device and update c test

### PR DESCRIPTION
**Pull request by jhdavis8**
This is a Fortran version of `test_target_teams_distribute_device`.

At present it passes xlf, but fails gfortran with the error: `libgomp: GOMP_target_enter_exit_data unhandled kind 0x05`

### CODE REVIEW

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_device.F90 [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    DO dev = 1, num_devices
+       !$omp target enter data map(to: a(1:N), b(1:N), c(1:N, dev:dev), &
+       !$omp& num_teams(dev:dev)) device(dev)
+    END DO
+
+    DO dev = 1, num_devices
+       !$omp target teams distribute map(alloc: a(1:N), b(1:N), &
+       !$omp& c(1:N, dev:dev), num_teams(dev:dev)) device(dev)
+       DO x = 1, N
+          num_teams(dev) = omp_get_num_teams()
```
Comment (josemonsalve2) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_device.F90]: data race

File: tests/4.5/target_teams_distribute/test_target_teams_distribute_device.F90 [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    END DO
+
+    DO x = 1, num_devices
+       num_teams(x) = 0
+       errors(x) = 0
+    END DO
+
+    DO dev = 1, num_devices
+       !$omp target enter data map(to: a(1:N), b(1:N), c(1:N, dev:dev), &
+       !$omp& num_teams(dev:dev)) device(dev)
```
Comment (spophale) [inline tests/4.5/target_teams_distribute/test_target_teams_distribute_device.F90]: num_teams is not a valid clause for "target enter data"

Comment (jhdavis8) [reply tests/4.5/target_teams_distribute/test_target_teams_distribute_device.F90]: I think in this pragma `num_teams` is actually the name of an array being mapped to the device (hence the `dev:dev`) to record the number of teams used on a particular device. I can rename the array if it is too misleading.

**Comment by jhdavis8** [comment]
I think this branch was made from an old new_test branch for a different test, rather than master, so I need to fix it before this is merged.

**Pull request closed without merging**